docs/security.md

```diff
@@ -0,0 +1,64 @@
+## Security
+
+Metadata application uses Spring OAuth2 and Keycloak authorization to secure REST APIs. W.r.t OAuth2 standards mappings are
+   - Authorization server: Keycloak server
+   - Resource server: Metadata Application
+   - Resource owner: Keycloak users with allowed role
+   - Client: Application calling Metadata REST APIs
+
+Endpoints
+
+   - / and /swagger-ui.html can be accessed by ananymous user. Ex: http://localhost:8085/ and http://localhost:8085/swagger-ui.html
+   - All other endpoints: 
+        - Read(GET) operation can be issued by any authenticated user with keycloak. The user can have any role assigned
+        - Update operation (POST, PUT, PATCH, DELETE) can be done by users only with “SERVICE_OPERATOR” role(described later)
+
+
+**Building metadata API application**
+
+   - Enter keycloak public key value in Application.properties “security.oauth2.resource.jwt.key-value”. It should have a value like “-----BEGIN PUBLIC KEY-----\nMIIBIjANBgkqhkiG9wxxxxxxxxxxxxxxxxxxxffjZs/D+e9A56XuRJSQ9QIDAQAB\n-----END PUBLIC KEY-----“
+   - This can be obtained from Keycloak Realm settings->Keys->Public key at the right-most column
+  
+ 
+**Configuring keycloak**
+  
+1. Setup and configure keycloak server with realm, client, role, mapper and user
+2. Consider following are configured in keycloak as an example
+     - Keycloak server running at: “http://localhost:8080” 
+     - Relam: “securemetadata”
+     - Client id: “secure-client”
+     - Role: “SERVICE_OPERATOR”. This role name is mandatory. It could be either client role or realm role.
+     - User name Mapper: Keycloak returns access token in JWT. Logged in username is returned in claim “preferred\_username” in access token. Spring Security OAuth2 resource Server expects username in claim “user\_name”. Hence a mapper to map logged in username to a new claim named user_name is required
+     - User: “svc_opr”
+     - Redirect URL in client settings: “http://localhost:8085” (Metadata application)
+3. Get all security endpoints from OpenID configuration URL "http://localhost:8080/auth/realms/securemetadata/.well-known/openid-configuration". 
+      - It should contain several endpoints and other information, which should appear like
+      >"issuer": "http://localhost:8080/ega/ampt2d/auth/realms/securemetadata" <br>
+      >“authorization_endpoint": "http://localhost:8080/ega/ampt2d/auth/realms/securemetadata/protocol/openid-connect/auth" <br>
+      >"token_endpoint": "http://localhost:8080/ega/ampt2d/auth/realms/securemetadata/protocol/openid-connect/token" <br>
+      >"token\_introspection_endpoint": "http://localhost:8080/ega/ampt2d/auth/realms/securemetadata/protocol/openid-connect/token/introspect" <br>
+      >"userinfo_endpoint": "http://localhost:8080/ega/ampt2d/auth/realms/securemetadata/protocol/openid-connect/userinfo"
+
+   
+**Get authorization token from keycloak server using postman**
```

Comment:
Again I think this is too much information, and the URLs are misleading because this service won't be deployed for the EGA in a while. Maybe this could be turned into a couple of points like the following:
```suggestion

## Testing Keycloak
1. Get authorization token from the Keycloak server using Postman (for testing)
2. In order to send the token
2.1. From Swagger: click the `Authorize` button and enter "Bearer <token obtained>"
2.2. From a REST client: include header "Authorization" with value "Bearer <token obtained>"
```
Everything below this point can then be dropped. If people experience trouble there are plenty of pages to check literally searching for "Get authorization token from keycloak server using postman".

Reply:
I agree with @cyenyxe. I guess Readme can contain only how to get authorization and how to test the application like provided in this. eg: https://github.com/EBIvariation/metadata-ws/blob/keycloakOauth/README.md. All other info regarding the setup of Keycloak can be put in confluence or the ticket.

Reply:
Should the section heading be `Testing keycloak`?